sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/authentication/ContainerRegistryCredentialsPolicy.java

```diff
@@ -84,6 +85,45 @@ public ContainerRegistryCredentialsPolicy(ContainerRegistryTokenService tokenSer
             });
     }
 
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
+            return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
+        }
```

Comment:
Do we need this check here? Should we just let the call to AAD throw if the `http` protocol is not supported?